### ISSUE TITLE
docker_container with pull: true will always report changed

### ISSUE THREAD

**Issue by jcassee** [ansible/ansible#19549]
_From @jcassee on December 2, 2016 16:51_

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
docker_container

##### ANSIBLE VERSION
```
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION
*none*

##### OS / ENVIRONMENT
Ubuntu 16.04

##### SUMMARY
When using `pull: yes`, the action always reports changed, even when the image is not changed (and so the container is not restarted). The old `docker` module would report ok in this case.

##### STEPS TO REPRODUCE
Run this playbook twice:

```
- hosts: localhost
  tasks:
    - docker_container:
        name: test
        image: alpine
        pull: yes
        command: sleep 10000
```

##### EXPECTED RESULTS
I would expect the second run to report *ok* for the task.

##### ACTUAL RESULTS
The task reports *changed* on the second run.

```
$ ansible-playbook test.yml
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [docker_container] ********************************************************
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0   

$ ansible-playbook test.yml
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [docker_container] ********************************************************
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0   

```

_Copied from original issue: ansible/ansible-modules-core#5804_

**Comment by jcassee** [comment]
_From @ansibot on December 2, 2016 17:19_


@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jcassee** [comment]
_From @ansibot on December 9, 2016 19:46_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by zeitounator** [comment]
Any progress on this one ?

**Comment by jgeusebroek** [comment]
Interesting, I came across the same behaviour. I will have a look at the plugin code later on, and see if I can fix it.

**Comment by AndreaGiardini** [comment]
Am I wrong or here https://github.com/ansible/ansible-modules-core/blob/devel/cloud/docker/docker_container.py#L1743 we should have an `and` instead of `or`?

**Comment by chouseknecht** [comment]
Fixed by #21508
